src/Sarif/Schemata/Sarif.schema.json

```diff
@@ -355,6 +382,60 @@
         }
       }
     },
+    "notification": {
+      "type": "object",
+      "description": "Describes a condition relevant to the tool itself, as opposed to being relevant to a file being analyzed by the tool.",
+      "properties": {
+        "id": {
+          "description": "A stable, unique identifier for the condition that was encountered.",
+          "type": "string"
+        },
+
+        "ruleId": {
+          "description": "The stable, unique identifier of the rule (if any) to which this notification is relevant.",
+          "type": "string"
+        },
+
+        "message": {
+          "description": "A string that describes the condition that was encountered.",
+          "type": "string"
+        },
+
+        "level": {
+          "description": "A value specifying the severity level of the notification.",
+          "default": "warning",
+          "enum": [ "note", "warning", "error" ]
+        },
+
+        "time": {
+          "description": "The date and time at which the analysis tool generated the notification.",
+          "type": "string",
+          "format": "date-time"
+        },
+
+        "exception": {
+          "description": "The runtime exception, if any, which caused this notification.",
+          "$ref": "#/definitions/exception"
+        },
+
+        "properties": {
+          "description": "Key/value pairs that provide additional information about the notification.",
+          "type": "object",
+          "default": { }
+        },
+
+        "tags": {
+          "description": "A set of distinct strings that provide additional information about the notification.",
```

Comment:
It reads right either way: "A (set of strings) that provides..." or "A set of (strings that provide)...". I had in mind the latter. Leaving.

---
In reply to: [61636809](https://github.com/Microsoft/sarif-sdk/pull/229#discussion_r61636809) [](ancestors = 61636809)